x/ibc/04-channel/keeper/packet.go

```diff
@@ -442,6 +442,7 @@ func (k Keeper) AcknowledgementExecuted(
 			sdk.NewAttribute(types.AttributeKeySrcChannel, packet.GetSourceChannel()),
 			sdk.NewAttribute(types.AttributeKeyDstPort, packet.GetDestPort()),
 			sdk.NewAttribute(types.AttributeKeyDstChannel, packet.GetDestChannel()),
+			sdk.NewAttribute(types.AttributeKeyData, string(packet.GetData())),
```

Comment:
I'm not entirely sure how helpful will be to emit the plain bytes as events since you will have to unmarshal them anyways. What is the use case you have in mind here @ancazamfir?

Reply:
I don't really need them for the relayer but they are showing in the send and receive packet events (not sure why they were put there in the first place) so I added this here for consistency.

Reply:
@alexanderbez could someone bloat the events by having a really large packet data? If so we should remove this event. cc: @AdityaSripal

Reply:
The reason we are emitting `packet.GetData()` in events in the first place is to remove the need to store the unhashed packet data on chain (we only store the hash). The relayer gets the packet data from the events and nowhere else.

So packet data **must** get emitted as an event attribute in `SendPacket`. It could be removed from `ReceivePacket` and `AcknowledgePacket` as far as I know. I don't know why we would need them apart from consistency in events you get from packet msg types.

It's a tradeoff between consistency and unnecessarily emitting packet data in `ReceivePacket` and `AcknowledgePacket`. @ancazamfir, is the packet data consistency important for having a clean relayer implementation?

Reply:
We can deal with some exceptions in the relayer. I also prefer not to include big blobs of data in the events if they are not needed.